deployments/helm/KubeArmorOperator/templates/deployment.yaml

```
@@ -2,7 +2,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ .Values.kubearmorOperator.name }}
-  namespace: {{ .Release.Namespace }}
+  namespace: {{ .Values.namespace | default "kubearmor" }}
```

IMO `namespace` in values is easy to overlook because most people are used to setting namespace with the `-n` flag. So this feels like unconventional.
How about to fallback to kubearmor NS we do something like:
```suggestion
  namespace: {{- if eq .Release.Namespace "default" }} {{ printf "kubearmor" | trim }} {{- else }} {{ .Release.Namespace }} {{- end }}
```

---

> IMO namespace in values is easy to overlook because most people are used to setting namespace with the -n flag. This will make the Helm -n flag useless which feels unconventional.


I didn't understand. What my understanding is that the default namespace to be used will be `kubearmor` and if a user defines any other ns using `-n` flag then that namespace will be used. Am I missing something?

---

@Ankurk99  Variable references prefixed with `.Values` are used to refer to variables that can be specified by user. `values.yaml` is intended to be the place where charts define these values so that the users can reference it (through docs or values.yaml) and also so that charts can set defaults for the values. This is the accepted best practice.

The `-n` flag translates internally to `.Release.Namespace` built-in variable. It is used to specify namespace that a chart should be installed into.
Ref - https://helm.sh/docs/chart_template_guide/builtin_objects/#helm

What I'm saying is instead of adding a new variable for namespace, let's use the already present built-in namespace object.

---

Got the point. I had an interesting conversation with Achref, and he made a good point. What if user actually wants to apply the kubearmor-operator in the `default` ns? 
I think we should change the ns value to be fetched from Release.Namespace and mention the default ns with the `-n` flag in the installation itself.

---

Sounds good :+1: